src/openssl.c

```diff
@@ -48,6 +48,21 @@
 
 #if __APPLE__
 #include <mach/mach_time.h> /* mach_absolute_time() */
+#define HAVE_ARC4RANDOM
+#endif
+
+#if defined(__FreeBSD_kernel__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
```

Comment:
According to https://sourceforge.net/p/predef/wiki/OperatingSystems/ you need to `#include <sys/param.h>` to define `BSD` symbol macro.